### ISSUE TITLE
Doesn't work with aliases

### ISSUE THREAD

**Issue by CaryJ** [tillig/ps-bash-completions#5]
I modified PSBashCompletions.psm1 to get it to work with aliases.  To do this, I read in my .alias.sh file as an init-file before processing the completion script.  I could submit my changes as a PR, but this might not be generic enough for other people's use.  At least it is working for me, now.
```
#  Write-Verbose "Completion command = &`"$bash`" `"$bashBridgeScript`" `"$bashCompletionScript`" `"<url-encoded-command-line>`""
  Write-Verbose "Completion command = &`"$bash`" --initfile `"~/.alias.sh`" -i `"$bashBridgeScript`" `"$bashCompletionScript`" `"<url-encoded-command-line>`""

#    $result = (&"$bash" "$bashBridgeScript" "$bashCompletionScript" "$encodedCommand")
    $result = (&"$bash" "--init-file" "~/.alias.sh" "-i" "$bashBridgeScript" "$bashCompletionScript" "$encodedCommand")
```

In my PowerShell profile, I added
```
Set-Alias kc kubectl
Register-BashArgumentCompleter kc $HOME\PsBashCompletions\kubectl-completions.sh
```
And, in my .alias.sh, I have to have the same alias I'm trying to configure in PowerShell, as well as a completion for it:
```
alias kc='kubectl'
complete -o default -F __start_kubectl kc
```

**Comment by tillig** [comment]
Can you describe more about what's not working? That is, it appears you have a bash alias file... but this is for PowerShell supporting basic completions that would otherwise normally only run in bash.

That is, at a PowerShell prompt, typing

`kubectl g<TAB>`

and getting a completion. It appears you have some sort of modification to add an alias to bash... but why? Is there a corresponding alias in PowerShell? Is this PowerShell-core-on-bash?

Did you try just adding a second registration for the alias, like

`Register-BashArgumentCompleter "kc" C:\completions\kubectl_completions.sh`

? If so, what happened?

**Comment by CaryJ** [comment]
I first tried with just a powershell alias
`Set-Alias kc kubectl`
but the completions didn't work with that.  I then tried, as you suggest,
`Register-BashArgumentCompleter kc C:\completions\kubectl_completions.sh`
But, completions on the alias still didn't work.  Next, I tried adding the corresponding alias in bash
`alias kc='kubectl'`
And discovered that completions didn't work in bash, even.  Some quick googling on how to get aliases working with bash completions led me to
`complete | grep kubectl`
to find the proper completion command, and adding that to my .bashrc, but with `kc` instead of `kubectl`
`complete -o default -F __start_kubectl kc`
Now, completions started working for `kc` from bash, but still not from PowerShell.

I assumed when you start bash from the PowerShell completion script that it was skipping reading in any startup rc or alias files, which I found to be true.  Since it could slow down the completions to read in the entire startup scripts, I pulled into a separate script just the alias and completion command that I want to work from PowerShell, also, and added those to `.alias.sh`, and changed `PSBashCompletions.psm1` to start bash with that init file.

Perhaps there can be an easier way to register aliases for completion, and have a different script generated that `PSBashCompletions.psm1` would run.  For example, `Register-BashAliasCompleter kc C:\PsBashCompletions\kubectl-completions.sh` would be similar to `Register-BashArgumentCompleter`, but add in the bash alias and complete commands into the script sent to bash.

**Comment by tillig** [comment]
Hmmm. This is an interesting thing that starts getting complicated really quickly. It'd be nice if the completer could hook into the native PowerShell alias thing so it "just works."

bash really wants just the command name - `kubectl` - without anything else. That is, no path (`/bin/kubectl`) and no extension (`kubectl.exe`).

When you `Set-Alias kc kubectl` it...
- Expands `kubectl` to locate the actual executable:
  - Name: `kubectl.exe`
  - Path: `C:\Program Files\Docker\Docker\Resources\bin\kubectl.exe`
- Adds an alias that points to that exact executable.

It's able to figure that mapping out because it has the rules of searching through the path and then looking at the `PATHEXT` environment variable to determine which file extensions are considered executable.

We could potentially try to reverse that in the module _before_ running the expansion.

- `$a = Get-Alias <aliasname>` to see if it's an alias. If it is...
- `$a.ResolvedCommand.Name` will be the executable, like `kubectl.exe`. If that value exists...
- Go through the list of `PATHEXT` values and remove the first one from the end of the command name that we find. `kubectl.exe` becomes `kubectl`.
- Now run the expansion on the result.

If we find there's no alias or the alias doesn't fit into the above logic, just run the expansion as normal. That'd ostensibly allow you to use a PowerShell alias for a command without having any bash modifications, and both the alias and the original command should work.

What do you think?

**Comment by CaryJ** [comment]
I like that approach - it definitely sounds like it could work!

**Comment by tillig** [comment]
Fixed in 1.2.0. You'll have to run `Register-BashArgumentCompleter` for the alias directly. If you want to use it for both `kc` and `kubectl` you'll need to register the completer for both. But I did test it out with that exact scenario and it seems to work:

```PowerShell
Set-Alias kc kubectl
Register-BashArgumentCompleter kubectl .\kubectl_completions.sh
Register-BashArgumentCompleter kc .\kubectl_completions.sh
```

I can do `kc c<TAB>` and get completions; and I can do `kubectl c<TAB>` and get the same correct completions.
